labgrid/driver/power/eg_pms2_network.py

```diff
@@ -0,0 +1,73 @@
+import re
+import requests
+
+from ..exception import ExecutionError
+
+# This driver implementes a power port for the EG_PMS2_LAN & EG_PMS2_WLAN
+# devices, it works through a simple HTTP interface, that requires a login.
+# Driver has been tested with:
+# * EG_PMS2_LAN
+
+# The login was successful when we can locate a Log out button
+LOGIN_SUCCESS_REGEX = str("<div class=\"boxmenuitem\"><a href="
+                          "\"login\\.html\">Log Out</a></div>")
+# Search for a string similar to: "var sockstates = [1,1,1,1];" and create
+# a match group out of the numbers within the square brackets.
+SOCKSTATES_REGEX = str("var\s+sockstates\s+=\s+"
+                       "\[([01]),([01]),([01]),([01])\];")
+
+def login(func):
+    """
+    Decorator automating login and logout from the Energenie web interface
+    """
+    def wrapper(host, port, index, value):
+        base_url = f"http://{host}:{port}"
+
+        login_url = f"{base_url}/login.html"
+        # Login - use the default password 1, because labgrid doesn't support
+        # password encryption, modifying the password doesn't secure the device
+        # as the password would be stored as plain-text.
+        try:
+            response = requests.post(login_url, data={'pw': 1})
+        except requests.exceptions.ConnectionError:
+            raise ExecutionError(f"Device not found at {host} or the network "
+                                 "interface of the device is not active "
+                                 "(press the reset button on the device)")
+        if response.status_code != 200 or not re.search(LOGIN_SUCCESS_REGEX,
+                                                        response.text):
+            raise ExecutionError("Login to Energenie web interface failed")
+
+        result = func(host, port, index, value)
+
+        # Logout
+        response = requests.get(login_url)
+        if response.status_code != 200:
+            raise ExecutionError("Logout from Energenie web interface failed")
+        return result
+    return wrapper
```

Comment:
Wouldn't two helper functions (login/logout) instead of the decorator be simpler?

Reply:
Hey Jan,

I guess they would be sufficient, I just applied my personal taste in this code and I am not mad if we go another route.
I just felt that...
```python
@login
def power_get(host, port, index):
    ...
    response = requests.get(status_url)
    ...
```
... concentrates more on the action that we are trying to do which is getting the status from the device. The login and logout felt more like a required extra, which can be abstracted out of the specific code into a decorator.
The alternative would be:
```python
def power_get(host, port, index):
    ...
    login(base_url=base_url)
    response = requests.get(status_url)
    logout(base_url=base_url)
    ...
```
Which from my point of view is a lot more verbose, so I prefer the decorator, but again no strong opinions, if simple helper functions fit more into the coding style of labgrid then I will change my code.

Greetings,
Sebastian

Reply:
@jluebbe 
Hey just wanted to check in, I would rebase this commit to the current tree, do you want me to refactor the code as well?

Greetings,
Sebastian